feat: add how-to for deploying GLAuth and SSSD as IAM stack

NucciTheBoss opened the pull request:
# Pre-submission checklist

 * [x] I read and followed the [CONTRIBUTING guidelines](../CONTRIBUTING.md).
 * [x] I have insured that the documentation tests complete successfully.

## Summary of Changes

[//]: # (Please summarize your commits here. For any complex or contenious changes, please also provide justifications.)

This PR adds a how-to for deploying GLAuth and SSSD as the IAM stack for Charmed HPC. It adds both CLI and Terraform instructions for deploying and integrating the services with Slurm, and shares next steps for managing users via SQL and and accessing the Postgres database provided by the `postgres-k8s` charm.

Other changes:

- I enabled substitutions within our documentation. We can now define aliases like `iam` in _substitutions.yaml_ and Sphinx will render the content as __{abbr}\`IAM (Identity and Access Management)\`__ within the page.
- I embedded reusable content such as Terraform plans for deploying GLAuth and SSSD and terminal output for showing what the output of certain commands like `juju status` will be.
- I removed the unused `ast` import in _conf.py_.

#### Related Issues, PRs, and Discussions

[//]: # (Please link to related issues, pull requests, and discussions here - especially corresponding code PRs. If your PR has no related issues, PRs, or discussions, please provide a justification for this PR here instead.)

* Closes #29

AshleyCliff commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+::::{tab-item} CLI
+:sync: cli
+
+First, use `juju switch`{l=shell} to switch from the `iam` model in your
+`charmed-hpc-k8s` Kubernetes cloud to the `slurm` in your `charmed-hpc`
+machine cloud:
```
Not sure model is the right word but some noun is needed here.

NucciTheBoss replied:
Yep, `model` is correct!

AshleyCliff commented on howto/setup/index.md:
```diff
@@ -5,6 +5,7 @@ the Slurm workload manager or a shared filesystem.
 
 - {ref}`howto-setup-deploy-slurm`
 - {ref}`howto-setup-deploy-shared-filesystem`
+- {ref}`howto-setup-deploy-glauth`
```
Could you include something like 'and GLAuth for Identity and Access Management (AIM)' to the description right above here?

NucciTheBoss replied:
Great idea!

NucciTheBoss replied:
Address in https://github.com/charmed-hpc/docs/pull/51/commits/ede23c9d5ba676d1bf1d90a527f253f1a5fe0730

AshleyCliff reviewed (changes requested):
Great work, there's tons of very involved details here and it's laid out very well! I've added some comments and suggestions but they're pretty minor overall.

NucciTheBoss commented on reuse/substitutions.yaml:
```diff
@@ -0,0 +1 @@
+iam: "{abbr}`IAM (Identity and Access Management)`"
```
Unfortunately there's not much I can do here to make the tooltip rendering text faster. The underlying HTML syntax uses the [`<abbr>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/abbr) tag, so your browser controls how fast the tooltip text is brought up, and how you interact with the element on the page.

Imagine the extra curious to know what `IAM` actually means will hover for a little longer, and they'll possibly be behooved to check out the "Glossary" reference page to get a better understanding of terms that they are not familiar with.

AshleyCliff replied:
Depending on how the Glossary gets implemented, we might eventually just have these link to their glossary entry rather than use the tooltip. I really want to like it, but the lag in rendering was rough.

NucciTheBoss commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+1. Using the [Juju CLI client](https://juju.is/docs/juju/juju-client).
+2. Using the [Juju Terraform client](https://juju.is/docs/juju/terraform-juju-client).
+
+If you want to use Terraform to deploy GLAuth and SSSD, see the
+[Install and manage the client (terraform juju)](https://juju.is/docs/juju/install-and-manage-the-client)
+how-to in the Juju documentation for additional requirements.
```
Hmm... it looks like the Terraform client documentation was moved. I'll need to update these links within the documentation as a whole.

NucciTheBoss replied:
The Terraform Juju provider has it's own documentation site now: https://canonical-terraform-provider-juju.readthedocs-hosted.com/en/latest/howto/manage-terraform-provider-juju/

NucciTheBoss commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+
+With the `juju_offer` and `juju_integration` resources declared, and external
+data sources loaded, you are now ready to connect SSSD to GLAuth using
+your `connect-sssd-to-glauth-.tf` plan: You can expand the dropdown below to
```
Good catch!

NucciTheBoss replied:
Addressed in https://github.com/charmed-hpc/docs/pull/51/commits/e760bb85735921417a6287e5b65aea1a27017d64

NucciTheBoss commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+---
+
+(howto-setup-deploy-glauth)=
+# How to deploy GLAuth and SSSD
```
Depends who our target reader is. System administrators should definitely know what is SSSD as it's the standard for enrolling machines with directory services such as Active Directory or OpenLDAP, but GLAuth will likely be a new concept as it comes from the Cloud-Native world.

I think it's fine for the ToC tree _for now_ since we'll more than likely be adding further identity implementations like Entra ID, but right now I don't want to abstract the title since I want to be explicit about what services you will be deploying in the Kubernetes cloud.

"How to deploy an identity stack" is too vague to me as everyone has their own implementation they might want to ship, and they might be misled with the "illusion of choice" if we abstract the title but your only options are really just GLAuth and SSSD.

NucciTheBoss commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+
+Now declare the following external data sources in your deployment plan. These
+data sources make Terraform aware of your pre-existing `iam` and `slurm` models,
+and the `glauth-k8s` and `sssd` applications:
```
Addressed in https://github.com/charmed-hpc/docs/pull/51/commits/e760bb85735921417a6287e5b65aea1a27017d64

NucciTheBoss commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+:::
+
+Now run the following set of commands to integrate GLAuth and the other
+applications together with `juju integrate`{l=shell}:
```
Addressed in https://github.com/charmed-hpc/docs/pull/51/commits/e760bb85735921417a6287e5b65aea1a27017d64

NucciTheBoss commented on howto/setup/deploy-glauth.md:
```diff
@@ -0,0 +1,357 @@
+## Prerequisites
+
+- An active [Slurm deployment](#howto-setup-deploy-slurm) in your [`charmed-hpc` machine cloud](#howto-initialize-machine-cloud)
+- The [Juju CLI client](https://juju.is/docs/juju/install-and-manage-the-client) installed on your machine
```
Addressed in https://github.com/charmed-hpc/docs/pull/51/commits/e760bb85735921417a6287e5b65aea1a27017d64 and https://github.com/charmed-hpc/docs/pull/51/commits/ab3fdde75823b101049b970b34dae4718e720ec9

NucciTheBoss commented:
@AshleyCliff ready for round two.

One thing to note is that I updated the Terraform sections to use subdirectories `glauth/`, `sssd/`, and `connect-sssd-to-glauth/` because Terraform does not accept unique filenames. The `-chdir` flag will direct Terraform to change into the relevant subdirectory when applying the plans for deploying GLAuth and SSSD. I also added substitutions for the plan names so it's easier to change them in the future should be possibly modify how the Terraform plans are put together in the howtos.

Let me know if you have any questions!

AshleyCliff commented:
Looks good! Merging now.
